qdb/tracer.py

```diff
@@ -408,33 +457,37 @@ def user_call(self, stackframe, arg):
 
     def user_line(self, stackframe):
         self.setup_stack(stackframe, None)
-        self.cmd_manager.send_watchlist()
-        self.cmd_manager.send_stack()
-        self.cmd_manager.next_command()
+        bound_cmd_manager = self.bound_cmd_manager
+        bound_cmd_manager.send_watchlist()
```

Comment:
this only happens in 3 places I think; it makes it much more clear what messages are actually being sent if they are explicitly referenced.